src/vs/workbench/contrib/customEditor/browser/commands.ts

```diff
@@ -172,3 +172,54 @@ MenuRegistry.appendMenuItem(MenuId.CommandPalette, {
 		model.redo();
 	}
 }).register();
+
+(new class ToggleCustomEditorCommand extends Command {
+	public static readonly ID = 'editor.action.customEditor.toggle';
+
+	constructor() {
+		super({
+			id: ToggleCustomEditorCommand.ID,
+			precondition: CONTEXT_HAS_CUSTOM_EDITORS,
+			kbOpts: {
+				primary: KeyMod.CtrlCmd | KeyCode.KEY_E,
+				weight: KeybindingWeight.EditorContrib
+			}
+		});
+	}
+
+	public runCommand(accessor: ServicesAccessor): void {
+		const editorService = accessor.get<IEditorService>(IEditorService);
+		const activeControl = editorService.activeControl;
+		if (!activeControl) {
+			return;
+		}
+
+		const activeGroup = activeControl.group;
+		const activeEditor = activeControl.input;
+		if (!activeEditor) {
+			return;
+		}
+
+		const targetResource = activeEditor.getResource();
+		if (!targetResource) {
+			return;
+		}
+
+		const customEditorService = accessor.get<ICustomEditorService>(ICustomEditorService);
+		const viewIDs = customEditorService.getContributedCustomEditors(targetResource);
+
+		const activeCustomEditor = customEditorService.activeCustomEditor;
+		let toggleView = 'default';
+		if (!activeCustomEditor) {
+			if (viewIDs && viewIDs.length) {
+				toggleView = viewIDs[0].id;
+			}
+			else {
+				return;
+			}
+		}
+
+		activeEditor.dispose();
+		customEditorService.openWith(targetResource, toggleView, undefined, activeGroup);
```

Comment:
It would be better to use `IEditorService.replaceEditors` to do the swap. That way, the new editor is kept in the same tab order as the one it replaces

Note that `replaceEditors` requires editor inputs but I believe you can use the `ICustomEditorService.createInput` method to create these

Reply:
Working with `replaceEditors` seem to work flawlessly when toggling from default to custom, but my editor tab seems to hang whenever I try to toggle from custom back to default.

Upon logging the `typeId` of the `EditorInput`, I found that the ID for the default editor—when testing toggles with README.md in `test/`— was "`workbench.editors.files.fileEditorInput`", while the ID for the replacement Input upon calling `ICustomEditorService.createInput` is "`workbench.editors.webviewEditor`". I feel like that should not matter, but is there possibly something that I'm missing?

I'm calling the replace function like so:
```
let toggleView = defaultEditorId;
if (!activeCustomEditor) {
	const viewIDs = customEditorService.getContributedCustomEditors(targetResource);
	if (viewIDs && viewIDs.length) {
		toggleView = viewIDs[0].id;
	}
	else {
		return;
	}
}

const replInput = customEditorService.createInput(targetResource, toggleView, activeGroup);
editorService.replaceEditors([{
	editor: activeEditor,
	replacement: replInput,
}], activeGroup);
```

Reply:
Resolved. Imported `IInstantiationService` to instantiate an Input of type `FileEditorInput` when the destination view ID is the default editor ID.